core/primitives/src/epoch_manager.rs

```diff
@@ -83,11 +83,17 @@ pub struct AllEpochConfig {
     use_production_config: bool,
     /// EpochConfig from genesis
     genesis_epoch_config: EpochConfig,
+    /// Chain Id.
+    chain_id: String,
 }
 
 impl AllEpochConfig {
-    pub fn new(use_production_config: bool, genesis_epoch_config: EpochConfig) -> Self {
-        Self { use_production_config, genesis_epoch_config }
+    pub fn new(
+        use_production_config: bool,
+        genesis_epoch_config: EpochConfig,
+        chain_id: &str,
```

Comment:
Not related but it would be cool to remove `use_production_config` as a follow up. I don't remember why but it really annoyed me. 
Generally I love the idea of customizing EpochConfigs based on the chain. I don't know if the chain name is the best way to go about it but it is an improvement.

Reply:
Good idea, but better to do it in a separate PR.